### PR TITLE
18245: Add aws_guard_duty resource

### PR DESCRIPTION
**Pull request by CyrusJavan**

### CODE REVIEW

File: aviatrix/resource_aviatrix_aws_guard_duty.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+				Optional:     true,
+				Description:  "Scanning Interval",
+				Default:      defaultAwsGuardDutyScanningInterval,
+				ValidateFunc: validation.IntInSlice(validAwsGuardDutyScanningIntervals),
+			},
+		},
+	}
+}
+
+func marshalAwsGuardDutyInput(d *schema.ResourceData) *goaviatrix.AwsGuardDuty {
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_guard_duty.go]: Move this function to the bottom.

File: aviatrix/resource_aviatrix_aws_guard_duty.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+			return fmt.Errorf("could not enable AWS GuardDuty for enabled_account %#v: %v", account, err)
+		}
+		err = client.UpdateAwsGuardDutyExcludedIPs(account)
+		if err != nil {
+			return fmt.Errorf("could not set excluded IPs for enabled_account %#v: %v", account, err)
+		}
+	}
+	// Only 1 GuardDuty resource can exist per controller, so, the ID is static.
+	d.SetId(awsGuardDutyID)
+	return nil
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_guard_duty.go:86]: Should be "return resourceAviatrixGatewayRead(d, meta)"?

File: aviatrix/resource_aviatrix_aws_guard_duty.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+			if err != nil {
+				return fmt.Errorf("could not enable GuardDuty for account %#v: %v", account, err)
+			}
+			err = client.UpdateAwsGuardDutyExcludedIPs(account)
+			if err != nil {
+				return fmt.Errorf("could not edit GuardDuty excluded IPs for account %#v: %v", account, err)
+			}
+		}
+	}
+	return nil
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_guard_duty.go:158]: Should return resourceAviatrixGatewayRead(d, meta)?

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

**Comment by CyrusJavan** [comment]
@zhiqiangjin I added `defer Read(d, meta)` to both the Update and Create.

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**